src/content/docs/en/guides/typescript.mdx

```diff
@@ -34,6 +34,40 @@ Additionally, our templates include an `env.d.ts` file inside the `src` folder t
 /// <reference types="astro/client" />
 ```
 
+If you are not using VSCode, it is also advisable to [install the typescript plugin from npm](https://www.npmjs.com/package/@astrojs/ts-plugin):
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install @astrojs/ts-plugin
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm install @astrojs/ts-plugin
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add @astrojs/ts-plugin
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Once you have done this, add the following to your `tsconfig.json`:
+
+```json title="tsconfig.json"
+  "compilerOptions": {
+    "plugins": [
+      {
+        "name": "@astrojs/ts-plugin"
+      },
+    ],
+    "jsx": "react",
```

Comment:
```suggestion

```

Reply:
This is needed even when not using react in sublime text, can't speak for other editors

Reply:
What if you're using SolidJS and would then need `jsx: preserve`? If it's necessary to have a JSX entry here, let's explain why and what values.

Reply:
Any `jsx` setting should be fine. `preserve` is the safest one, since it'll work no matter which framework you're using.